src/fw_update.c

```diff
@@ -75,26 +78,60 @@ static enum golioth_status fw_write_block_cb(const struct golioth_ota_component
     return status;
 }
 
-static enum golioth_status golioth_fw_update_report_state_sync(struct golioth_client *client,
-                                                               enum golioth_ota_state state,
-                                                               enum golioth_ota_reason reason,
-                                                               const char *package,
-                                                               const char *current_version,
-                                                               const char *target_version,
-                                                               int32_t timeout_s)
+enum golioth_status golioth_fw_update_report_state_sync(struct fw_update_component_context *ctx,
+                                                        enum golioth_ota_state state,
+                                                        enum golioth_ota_reason reason,
+                                                        bool report_component_name,
+                                                        bool report_current_version,
+                                                        bool report_target_version)
```

Comment:
How about replacing 3 `bool` arguments with single `int` argument that would allow flags to be ORed together with `REPORT_COMPONENT_NAME | REPORT_CURRENT_VERSION | REPORT_TARGET_VERSION`?

The problem with `bool` arguments is that when passing them to function, it is very hard to know what those are about. It requires to look at function prototype to see that each `true` or `false` does. On the other hand passing flags (macros) directly to function allows to see what exactly those parameters are about.